.github/workflows/cd.yaml

```diff
@@ -32,3 +32,8 @@ jobs:
           # See https://docs.github.com/en/actions/reference/events-that-trigger-workflows#triggering-new-workflows-using-a-personal-access-token
           github-token: ${{ secrets.GU_GUARDIAN_CI_TOKEN }}
           npm-lockfile-version: 2
+
+      - if: ${{ github.ref == 'refs/heads/main' }}
+        name: run-project-generation
+        # This ensures that the newly released version of the library can still be used to generate a new project from scratch
+        run: npx @guardian/cdk@latest new --stack my-stack --app my-app
```

Comment:
The `new` command [installs dependencies](https://github.com/guardian/cdk/blob/bf89943f4ea655994088870d9caa4e08bd5e2089/src/bin/commands/bootstrap.ts#L133-L135) and [runs snapshot tests](https://github.com/guardian/cdk/blob/bf89943f4ea655994088870d9caa4e08bd5e2089/src/bin/commands/bootstrap.ts#L147-L149), so if the command exits successfully we should have a good level of confidence that the project generation tooling is working correctly.

Reply:
Because we have chosen to run this command post-release, it's happening as part of a workflow that is only triggered for `main` and `beta` branches anyway: https://github.com/guardian/cdk/blob/9918d03f7118d354d2accb0ebeb3bdef63c63180/.github/workflows/cd.yaml#L4-L6

This new step is using the latest _released_ version of the `cli` (via `npx`). The cli command itself is installing the latest _released_ version of the library from `npm` (i.e. the version that will have been released by earlier steps from this workflow) in order to generate a new `cdk` project. If we were running this step against other branches it wouldn't really make sense IMO, because we would be testing the version of the library that's been released, rather than the version that exists on the feature branch. 

As @akash1810 correctly points out:

> not every change to `main` produces a new release.

so technically running it against _every_ build of `main` will still mean that we will sometimes execute this command superfluously. I don't think the cost of that is too high because this build doesn't block anyone/anything.